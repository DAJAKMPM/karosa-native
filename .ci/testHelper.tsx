import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
