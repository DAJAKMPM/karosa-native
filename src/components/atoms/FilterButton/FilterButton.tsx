/**
 *
 * FilterButton
 * @format
 *
 */

import React, { FC } from "react";
import { Button } from "react-native-elements";
import type { PropsType } from "./types";
import FilterButtonStyles from "./styles";

const FilterButton: FC<PropsType> = (props) => {
  const { onPress, title, icon, buttonStyle } = props;

  const onClick = () => {
    onPress();
  };

  return (
    <Button
      onPress={onClick}
      title={title}
      iconRight={true}
      iconContainerStyle={FilterButtonStyles.iconContainer}
      buttonStyle={buttonStyle || FilterButtonStyles.bgNeutral}
      icon={icon}
      containerStyle={FilterButtonStyles.container}
      titleStyle={FilterButtonStyles.title}
      type={"outline"}
    />
  );
};

export default FilterButton;
