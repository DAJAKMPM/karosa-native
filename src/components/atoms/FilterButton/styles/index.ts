/**
 *
 * @format
 *
 */

import { theme } from "@app/styles";
import { StyleSheet } from "react-native";

const FilterButtonStyles = StyleSheet.create({
  bgNeutral: {
    borderColor: theme.colors.grey5,
    backgroundColor: theme.colors.grey5,
    borderWidth: 2,
  },
  bgClicked: {
    borderColor: theme.colors.green5,
    backgroundColor: theme.colors.grey5,
    borderWidth: 2,
  },
  title: {
    color: theme.colors.black,
    fontSize: 10,
    paddingLeft: 0,
  },
  container: {
    margin: 10,
  },
  iconContainer: {
    right: 5,
  },
});

export default FilterButtonStyles;
