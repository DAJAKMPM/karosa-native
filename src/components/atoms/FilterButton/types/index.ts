/**
 *
 * @format
 *
 */

<<<<<<< HEAD
 import { StyleProp, ViewStyle } from "react-native";
 import { IconNode } from "react-native-elements/dist/icons/Icon";
 
 export type PropsType = {
   onPress(): void;
   id?: number;
   title: string;
   icon?: IconNode | undefined;
   buttonStyle?: StyleProp<ViewStyle>;
   titleStyle?: StyleProp<any>;
   containerStyle?: StyleProp<any>;
 };
=======
import { StyleProp, ViewStyle } from "react-native";
import { IconNode } from "react-native-elements/dist/icons/Icon";

export type PropsType = {
  onPress(): void;
  id?: number;
  title: string;
  icon?: IconNode | undefined;
  buttonStyle?: StyleProp<ViewStyle>;
};
>>>>>>> a331fff... feat(productdetail modal): added some coding to highlight the variations selected
