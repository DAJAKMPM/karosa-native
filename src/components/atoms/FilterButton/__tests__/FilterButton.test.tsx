import React from "react";
import { noop } from "lodash";
import { renderWithWrapper } from "@app/ci/testHelper";
import FilterButton from "../FilterButton";

describe("FilterButton Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <FilterButton onPress={noop} title="test" />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
