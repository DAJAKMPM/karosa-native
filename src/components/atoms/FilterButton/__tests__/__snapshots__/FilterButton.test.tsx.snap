// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterButton Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        Array [
          Object {
            "margin": 10,
          },
          undefined,
        ],
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F2F3F7",
            "borderColor": "#F2F3F7",
            "borderRadius": 3,
            "borderWidth": 2,
            "flexDirection": "row-reverse",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontSize": 10,
              "paddingLeft": 0,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          test
        </Text>
      </View>
    </View>
  </View>
</View>
`;
